### PR TITLE
[e2e] use ci job id to name stacks

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Use job id instead of pipeline id to name pulumi stacks on the CI

### Motivation

Limit collisions on stacks across jobs running on the same pipeline. We currently ask to ensure each concurrent job has one stack with one unique name, this change would allow to run the same test suite with different flavours on the same pipeline

### CODE REVIEW

File: test/new-e2e/pkg/runner/ci_profile.go [hunk 1]
```diff
@@ -38,13 +38,14 @@ func NewCIProfile() (Profile, error) {
 	if err != nil {
 		return nil, fmt.Errorf("unable to get pulumi state password, err: %w", err)
 	}
+	// TODO move to job script
```
Comment (vboulineau) [inline test/new-e2e/pkg/runner/ci_profile.go:41]: Why? The goal of profiles + `secretStore` is to do exactly that.

**Review by vboulineau — APPROVED** [review]

File: test/new-e2e/pkg/runner/ci_profile.go (hunk 1 above)
Comment (pducolin) [reply test/new-e2e/pkg/runner/ci_profile.go:41]: I'm triggered when I see usage of `Setenv` in code as it is not thread-safe.

Why are we setting this variable here vs in the CI job definition? We should rather pass it to https://github.com/DataDog/datadog-agent/blob/pducolin/test-infra-move-to-file/test/new-e2e/pkg/utils/infra/stack_manager.go#L367

Comment (vboulineau) [reply test/new-e2e/pkg/runner/ci_profile.go:41]: A lot of Go code is not thread safe (all maps), and has no reason to be as only 1 goroutine will ever run it at the same time, which is the case here.

I'm not sure it works setting it inside the workspace, but if it does, it's indeed better to put it there.

Comment (pducolin) [reply test/new-e2e/pkg/runner/ci_profile.go:41]: From the [documentation](https://pkg.go.dev/github.com/pulumi/pulumi/sdk/v2/go/x/auto#EnvVars) it sounds like it does work, forgot to link it to my previous post.

Maps are expected to be not thread safe, `Setenv` like file access is rather unexpected, unless you are familiar with C implementation. So I avoid `Setenv` usage as it is considered a bad practice.

**Pull request merged**